puppetry/__init__.py

```diff
@@ -180,22 +181,34 @@ def sendSet(data):
     ''' Send a set request to the viewer
             data must be a dict
     '''
+
+    req_id = -1
     if _running:
         if isinstance(data, dict):
+            req_id = get_next_request_id()
             msg = { 'command':'set', 'data':data }
-            msg.setdefault('reqid', get_next_request_id())
+            msg.setdefault('reqid', req_id)
+
+            if 'reqid' in data:
+                if data['reqid'] == 'auto': #HACK: Echo request ID into message.
+                    data['reqid'] = reqid
```

Comment:
This and line 199 probably want to reference `req_id`? Or maybe better, change lines 188 and 190 (and 185 if you keep it) to `reqid`. As written, I think this would lead to a `NameError`.

That aside, I'm confused by having `reqid` both at the outer `{'command': ..., 'data':... }` level and within `data`. Unless `data['reqid']` is `'auto'`, they'll be different values. Which one is supposed to be used by the viewer?

I don't see any `sendSet()` calls in this PR, so I don't know how the returned `reqid` would be used, or whether any calls will in fact use `'auto'`.

For that matter, I don't see any references to `'reqid'` in the new report-related functions, so I still don't understand the intent of the new viewer code in the other PR.